### DISCUSSION TITLE
Producer and Cosumers  spans comes under different trace  Id for Ottrace Propagator

### DISCUSSION THREAD

**Discussion by pkishork** [open-telemetry/opentelemetry-java#4101] [Q&A]
I a using kafka  channel  in my project . implemented manual instrumentation using open telemetry 1.9.1 libraries  version . For all propagator except Ottrace propagator Producer and  consumer  spans come under same trace id  . But for Ottrace  comes under different trace id.  do we need add other  propagator along with ottrace  ? or for Ottrace  producer and consumer spans always come  under different trace id ? 
<img width="937" alt="ottrace" src="https://user-images.githubusercontent.com/97011611/150351686-43b2dc02-341a-4ec0-9675-94ba8879c137.png">

**Comment by jkwatson** [comment]
Unless you're needing to interop with OpenTracing, I probably wouldn't use the ottrace propagator at all. We'd need much more information about what you're doing to answer why you might not be getting proper propagation.

**Reply by pkishork** [reply]
in  our project   traceId of length 32 from producer side  but when context goes to consumer side  , OtTracePropagator :: inject  => changes the traced id length of 16 chars  
 https://github.com/open-telemetry/opentelemetry-java/blob/98fdf9b4bc9c038cce81d1e92664c2400915994f/extensions/trace-propagators/src/main/java/io/opentelemetry/extension/trace/propagation/OtTracePropagator.java

Thats why getting different  trace id  on consumer side  . so that  producer spans and consumer spans  shows  under different trace id on jaegar UI . 

what would be  solution for this . how can we  maintain   same trace id  on both side  from producer and consumer side  like any other format

**Reply by jkwatson** [reply]
Why are you using the ottrace propagator? It's only recommended for very narrow use-cases.
